Fix the docstring decoration

whitphx opened the pull request:
## Description

Seems like the document is not rendered as Markdown and backticks are not considered as inline code syntax. Instead, `{...}` is needed to use.

https://www.gradio.app/docs/slider
![CleanShot 2023-10-12 at 18 28 38@2x](https://github.com/gradio-app/gradio/assets/3135397/93e9d0c7-a1b3-4338-9ab0-3fc1382f9b9f)

If this fix is acceptable, I will also fix other docstrings which have the same problem.

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

abidlabs commented:
Hmm I don't see it taking effect in the website preview?

<img width="1030" alt="image" src="https://github.com/gradio-app/gradio/assets/1778297/0d7f0a3a-e936-4c9c-9cae-430a6f257e0f">

aliabd commented:
This actually won't do anything as is @whitphx, we only convert the curly braces to monospace in preprocessing/postprocessing sections of doc. It's done here: https://github.com/gradio-app/gradio/blob/e70805d54cc792452545f5d8eccc1aa0212a4695/js/_website/generate_jsons/src/docs/__init__.py#L100-L119

You should modify this code for it to work on the description as well (which I think is a great idea). Or I can do it if you'd like

aliabd commented:
this should do the trick

```python
def style_types():
    for mode in docs:
        for cls in docs[mode]:
            cls["description"] = cls["description"].replace("{",
                        "<span class='text-orange-500' style='font-family: monospace; font-size: large;' >"
                    ).replace("}", "</span>")
            for tag in [
                "preprocessing",
                "postprocessing",
                "examples-format",
                "events",
            ]:
                if tag not in cls["tags"]:
                    continue
                cls[tag] = (
                    cls["tags"][tag]
                    .replace(
                        "{",
                        "<span class='text-orange-500' style='font-family: monospace; font-size: large;' >",
                    )
                    .replace("}", "</span>")
                )

```

please also do it for other docstrings!

whitphx commented:
Aww, thank you, will modify the code as well.
Also will do it for other docstrings, thanks!

whitphx commented on js/_website/generate_jsons/src/docs/__init__.py:
```diff
@@ -97,9 +97,19 @@ def add_guides():
 def style_types():
     for mode in docs:
         for cls in docs[mode]:
+            cls["styled_description"] = convert_inline_code_syntax(cls["description"])
```
To avoid overwriting the existing `description` field, I added a new field `styled_description` for this purpose.
This is different from the `tags.*` field because they are not overwritten as follows.
```json
{
  "description": "<raw description text>",
  "styled_description": "<styled description text>",  // Added in this PR, because the original `description` field is at the top level.
  "tags": {
    "preprocessing": "<raw preprocessing text>"
  },
  "preprocessing": "<styled preprocessing text>"  // This field added by this function, avoiding overwriting the original `tags.preprocessing` field.
}
```

whitphx commented:
There are also different decoration markers exists such as `*<text>*`:
https://www.gradio.app/docs/annotatedimage
![CleanShot 2023-10-13 at 15 36 55@2x](https://github.com/gradio-app/gradio/assets/3135397/79f6a2f6-0d44-46a6-a0d1-0eab2010854e)
To convert it, we should update the syntax converter (`style_types()`). So, it also looks better to update it to deal with the backticks as the inline code syntax, instead of re-writing the docstrings?

aliabd reviewed (commented):
Nice work! 

A few things: 

* we should also apply the styling to fns. Add this to `style_types`:
```python
for fn in cls["fns"]:
                fn["styled_description"] = convert_inline_styles(fn["description"])
```
* I think you should just override description, that's totally fine. Otherwise, there are a few more spots in the website where you'll have to swap out `obj.description` for `obj.styled_description`. You did it in `js/_website/src/routes/[[version]]/docs/[doc]/+page.svelte` which is the main place, but we do have other one-off pages in the docs, so it will have to be swapped there too. 
```
js/_website/src/routes/[[version]]/docs/flagging/+page.svelte
js/_website/src/routes/[[version]]/docs/themes/+page.svelte
```
Also in 
```
js/_website/src/lib/components/FunctionDoc.svelte
``` 
where `fn.description` will need to be updated.

whitphx commented:
Thank you!

First of all I changed my mind and moved such HTML/CSS styling logics to the JS code (`$lib/text.ts`).

Then, fixed the files you mentioned as well 👍

abidlabs commented:
What's the status of this PR @whitphx @aliabd?

whitphx commented:
@abidlabs Waiting for the review, thanks!

aliabd reviewed (approved):
@whitphx sorry for the delay on this, I committed some changes that fixed errors and moved the logic from .svelte to .server.ts 

Everything looks great now!

freddyaboulton commented:
Going to merge since @aliabd made the necessary final fixes. Thanks @aliabd and @whitphx !!
